### DISCUSSION TITLE
Pocketbase SSL Issue When Running on Linux Server.

### DISCUSSION THREAD

**Discussion by EthanTuning** [pocketbase/pocketbase#2911] [Q&A] [answered]
I have had a Pocketbase backend running on a Linode Server for a couple of months now, serving as a backend for a few different experimental apps I am building. The server runs on the Debian Linux distro, and I set up a Linux service to serve Pocketbase, ensuring it restarts if the server has to restart.

![image](https://github.com/pocketbase/pocketbase/assets/17459105/cbe8b726-86d5-4514-b893-326211e67070)

Pocketbase has been running just fine with this configuration, for a while now and everything has been fine. Yesterday though, the API stopped responding as I was developing, with this response:

![image](https://github.com/pocketbase/pocketbase/assets/17459105/7969a234-c69c-4af5-8b17-5a8cc7e840aa)

So, I decided to try and access the Admin UI and it could not be reached:

![image](https://github.com/pocketbase/pocketbase/assets/17459105/8c51ee1e-d357-4720-89a7-3fb14e771399)

Then, I ssh'd into the server to see what I could find. I tried restarting my service: 

```
systemctl stop pocketbase
systemctl enable pocketbase.service
systemctl start pocketbase
```

Still was not working, so I manually served Pocketbase, so see the output and saw this when trying to hit the API or the Admin UI:

![image](https://github.com/pocketbase/pocketbase/assets/17459105/f19f4b94-09f2-4111-b3fb-535e0f1b6674)

So, seems like some sort of issue with this `autocert` Go package, which causes this TLS/HTTPS problem when trying to access the server.

Any thoughts on how I might mitigate this? Possibly a Linux setting? I keep restarting the server, restarting the Pocketbase service, and nothing seems to get it to work again.

Thanks!

**Comment by NotNullDev** [comment]
May be silly question, but are you accessing your server via domain name or directly from the IP address? I believe it is required to have domain name in order to acquire Let's Encrypt ssl certificate.

**Reply by EthanTuning** [reply]
Not silly at all, but yes, I am using a domain name. I did try serving Pocketbase and accessing it through the IP address as well to see if that worked actually, but realized that would probably not work anyway, to your point.

**Comment by ganigeorgiev** [comment, marked as answer]
I doubt that this is an issue with the `autocert` Go package, as it wouldn't explain why it worked previously (in addition the cert for the the live demo seems to work and renew fine).

How is your domain managed?
Have you done any DNS changes recently?
Are there any recent changes to the `/etc/hosts` file?

I'm not sure if this will help, but you can try to manually delete the `/pb_data/.autocert_cache/` directory and start again the application to force issuing a new certificate (note that there are [Lets's encrypt rate limits](https://letsencrypt.org/docs/rate-limits/)).

**Reply by EthanTuning** [reply]
No, no recent changes to `/etc/hosts`. But okay, I will try that deleting the `autocert_cache` next.

The domain is managed via a DNS zone file, that is hosted in GoDaddy, I believe.. Our company's IT guy manages it, I will ask again if any changes have been made recently, but he did say that nothing has changed. I may have to look at it directly though.

Is there something I could do to `/etc/hosts` file possibly? If deleting the `autocert_cache` does not work? 

Thank you.

**Reply by ganigeorgiev** [reply]
> Is there something I could do to /etc/hosts file possibly?

Usually the domain should be mapped to your server host address because when `--https=yourdomain.com:443` is used we try to bind to the ip behind yourdomain.com. But if the Go application was able to start then I guess your current configuration should be fine.

Is there any firewall/iptable rules/etc. in place that could block the TLS challenges?

Are there any extra redirects? When you run `curl -Ii yourdomain.com` is there a `Location` header that 302 redirects directly to the `https://yourdomain.com/` version?

Without access to at least the domain I'm not sure how to help you debug it. We haven't done any changes to the TLS generations in the recent PocketBase releases so I doubt that the issue is caused by a change in the code.

**Reply by EthanTuning** [reply]
Great info, thank you very much. 

I will dig some more and get to the bottom of this. But yeah, more I think about it, it has to be something with our domain configuration or something. I will let you know what I figure out.

Again, thank you, I appreciate the responses, and the hard work on this awesome tool!
